itou/approvals/models.py

```diff
@@ -241,24 +241,6 @@ def can_be_suspended_by_siae(self, siae):
             and not self.user.last_accepted_job_application.can_be_cancelled
         )
 
-    def suspend(self, start_at, end_at, siae, reason, reason_explanation, created_by):
-        """
-        Suspend current Approval.
-        """
-        if not self.can_be_suspended_by_siae(siae):
-            raise RuntimeError(_("Approval cannot be suspended by this SIAE."))
-        suspension = Suspension(
-            approval=self,
-            start_at=start_at,
-            end_at=end_at,
-            siae=siae,
-            reason=reason,
-            reason_explanation=reason_explanation,
-            created_by=created_by,
-        )
-        suspension.save()
-        return suspension
```

Comment:
À l'origine je voulais commencer à tout encapsuler dans le modèle. J'avais ajouté `def suspend()` pour la création afin d'éviter d'avoir à utiliser `.save()` dans les vues.

Mais du coup ça marche pas bien avec la logique des `ModelForm` (voir le paragraphe [*Breaking the contract*](https://www.dabapps.com/blog/django-models-and-encapsulation/)).

Mais j'ai quand même utilisé des `ModelForm` car ils déclenchent tout le mécanisme de validation avec `clean()` etc.

Du coup, il y a modification d'attributs et appels explicites à `save()` dans les vues mais ça permet d'économiser beaucoup de code.

Quelle est votre opinion ?

Reply:
Ok pour moi. Code plus succint au final.